src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1238,286 +1238,115 @@ void RequireDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionC
 
 		void MarkTypeForDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionContext, TypeDefinition typeDefinition, DynamicallyAccessedMemberTypes requiredMemberKinds)
 		{
-			if (requiredMemberKinds == DynamicallyAccessedMemberTypes.All) {
-				var sourceMethod = reflectionContext.SourceMethod;
-				reflectionContext.RecordRecognizedPattern (typeDefinition, () => _markStep.MarkEntireType (typeDefinition, includeBaseTypes: true, new DependencyInfo (DependencyKind.AccessedViaReflection, sourceMethod)));
-				return;
+			foreach (var member in DynamicallyAccessedMembersBinder.GetDynamicallyAccessedMembers (typeDefinition, requiredMemberKinds)) {
```

Comment:
Would be better to do such refactoring in separate PR

Reply:
Done in https://github.com/mono/linker/pull/1223.